vmm/src/lib.rs

```diff
@@ -1445,17 +1445,19 @@ impl Vmm {
             }
         }
 
+        let default_config = VmConfig::default();
```

Comment:
nit: You could use the `default_config` as the new VmConfig and only update the defaults when there is a new value in the `machine_config` parameter.

```rust
let mut updated_vm_config = VmConfig::default();
if let Some(ht_enabled) = machine_config.ht_enabled {
     updated_vm_config.ht_enabled = Some(ht_enabled);
}

....

// after updating all fields and doing all checks you can set self.vm_config to be updated_vm_config:
self.vm_config = updated_vm_config;
```

You can get rid of some useless copies with this approach, but probably it doesn't make a huge difference.

Reply:
+1
Also don't forget to guard against empty `PUT`

Reply:
@acatangiu that check is done when parsing the request in the API server. Do you mean we should also have it here?

Reply:
No, that's fine. (I hadn't checked the API server)

Reply:
Hi! I have updated my branch, please re-review, thanks!